src/Http/WebUtilities/src/FormPipeReader.cs

```diff
@@ -167,7 +167,7 @@ public FormPipeReader(PipeReader pipeReader, Encoding encoding)
                     if (!isFinalBlock)
                     {
                         // Don't buffer indefinately
-                        if (span.Length > KeyLengthLimit + ValueLengthLimit)
+                        if ((uint)span.Length > (uint)KeyLengthLimit + (uint)ValueLengthLimit)
```

Comment:
What happens if you use int.MinValue? Wouldn't casting to a uint cause overflow again here?

Reply:
Negative values are invalid anyways so I'm not concerned about them. MaxValue is not a good idea, but still in the valid value range.

Reply:
Do we throw in the form reader options if it is less than 0?

Reply:
I don't think we do: https://github.com/aspnet/AspNetCore/blob/master/src/Http/Http/src/Features/FormOptions.cs maybe we should?

Reply:
It's invalid regardless of if we throw. I'd only worry about adding checks for that if anyone ever reported trying to do it.